### PR TITLE
install.sh: Add the INSTALL_K3S_MIRROR variable to increase k3s binary file download speed

### PR DESCRIPTION
**Pull request by xiaoluhong**

### CODE REVIEW

**Comment by erikwilson** [comment]
This should probably be more generic, where we could say something like `INSTALL_K3S_MIRROR=cn`, or provide a custom URL. For official mirrors we should probably also publish from our build system.

**Comment by xiaoluhong** [comment]
@erikwilson  Okay, fine, correction completed. Thank you.

File: install.sh [hunk 1]
```diff
@@ -69,10 +69,16 @@ set -e
 #   - INSTALL_K3S_TYPE
 #     Type of systemd service to create, will default from the k3s exec command
 #     if not specified.
+#
+#   - INSTALL_K3S_MIRROR
+#     For Chinese users, set INSTALL_K3S_MIRROR=cn to use the mirror address to accelerate 
+#     k3s binary file download, and the default mirror address is mirror_k3s.rancher.cn
```
Comment (erikwilson) [inline install.sh]: The help text should be more generic.

File: install.sh [hunk 2]
```diff
@@ -69,10 +69,16 @@ set -e
...
+#
 
 GITHUB_URL=https://github.com/rancher/k3s/releases
 STORAGE_URL=https://storage.googleapis.com/k3s-ci-builds
 DOWNLOADER=
+INSTALL_K3S_MIRROR_URL=${INSTALL_K3S_MIRROR_URL:-mirror-k3s.rancher.cn}
```
Comment (erikwilson) [inline install.sh]: `INSTALL_K3S_MIRROR` should be able to take a url

File: install.sh [hunk 3]
```diff
@@ -372,6 +380,8 @@ installed_hash_matches() {
 download_binary() {
     if [ -n "${INSTALL_K3S_COMMIT}" ]; then
         BIN_URL=${STORAGE_URL}/k3s${SUFFIX}-${INSTALL_K3S_COMMIT}
+    elif [ "${INSTALL_K3S_MIRROR}" = cn ]; then
+        BIN_URL=${INSTALL_K3S_MIRROR_URL}/download/`echo ${VERSION_K3S} | sed 's/+/-/g'`/k3s${SUFFIX}
```
Comment (erikwilson) [inline install.sh]: Better to abstract the + -> - transformation to another variable

**Comment by davidnuzik** [comment]
@xiaoluhong May I ask why you closed this PR?

**Comment by davidnuzik** [comment]
Please review https://github.com/rancher/k3s/pull/1700 instead.

**Pull request closed without merging**